exercises/047_methods.zig

```diff
@@ -2,41 +2,44 @@
 // Help! Evil alien creatures have hidden eggs all over the Earth
 // and they're starting to hatch!
 //
-// Before you jump into battle, you'll need to know four things:
+// Before you jump into battle, you'll need to know three things:
 //
-// 1. You can attach functions to structs:
+// 1. You can attach functions to structs (and other "type definitions"):
 //
 //     const Foo = struct{
 //         pub fn hello() void {
 //             std.debug.print("Foo says hello!\n", .{});
 //         }
 //     };
 //
-// 2. A function that is a member of a struct is a "method" and is
-//    called with the "dot syntax" like so:
+// 2. A function that is a member of a struct is "namespaced" within
+//    that struct and is called by specifying the "namespace" and then
+//    using the "dot syntax":
 //
 //     Foo.hello();
 //
-// 3. The NEAT feature of methods is the special parameter named
-//    "self" that takes an instance of that type of struct:
+// 3. The NEAT feature of these functions is that if they take either
+//    an instance of the struct or a pointer to an instance of the struct
+//    then they have some syntax sugar:
```

Comment:
I'm not sure I understand what is meant by `then they have some syntax sugar`.  Can we make it more clear what the additional sugar is? (I assume it's the passing of the instance when we invoke the function?)

Reply:
Yes. I'm trying to say that if (their first parameter is `@This()`, `*@This()`, or `*const @This()`) then (syntax sugar: `thing.function(args)` -> `@TypeOf(thing).function(thing, args)`)

I'll try to clarify that without the need to understand `@TypeOf` though.